README.md

```diff
@@ -253,6 +253,93 @@ dependencies {
 
 With spring devtools on the classpath your application will automatically reload as you edit and build your java classes.
 
+## How the cftlib works
+
+The cftlib uses isolated classloaders to run multiple spring boot applications in a single Java Virtual Machine (JVM).
+
+```mermaid
+graph TD;
+    boot[Bootstrap classloader]-->app[Your app's classloader];
+    boot-->datastore[CCD data store classloader];
+    boot-->definition[CCD definition store classloader];
+    boot-->etc[etc...];
+```
+
+By running each spring boot application in its own classloader dependency conflicts are avoided; each application can have its own unique dependency set.
+
+### Project structure
+
+#### rse-cft-lib-plugin
+
+The cftlib Gradle plugin that configures the build of the consuming project, creating the build tasks, sourcesets etc.
+
+#### lib/
+
+The lib folder contains libraries that are published to the jitpack maven repository and are consumed as dependencies when running the cftlib.
+
+##### lib/bootstrapper
+
+An application that creates each of the necessary classloaders to run our spring applications and defines the Cftlib API (but not its implementation). 
+
+This project runs on the system classloader (meaning it is on the classpath to the JVM upon launch). Since the system classloader is parent to the isolated classloaders that run our applications, classes in this project are accessible to all running services.
+
+This project contains the 'control plane'; a coordination class invoked by the cftlib-agent (see next project).
+
+A consequence of being on the system classloader is that this project should be dependency free; any dependencies present on the system classloader would override those in the isolated classloaders leading to potential conflicts.
+
+##### lib/cftlib-agent
+
+Added to the classpath of each spring boot application that the cftlib runs, enabling the injection of new & custom functionality.
+
+For example, to coordinate the boot process a Spring boot event listener detects when each spring boot application is ready and reports it to the bootstrapper control plane.
+
+```mermaid
+graph BT;
+    boot[Bootstrap classloader <br> ControlPlane::appReady]; 
+    app[App <br> libagent]--ApplicationReadyEvent-->boot;
+    data[Datastore <br> libagent]--ApplicationReadyEvent-->boot;
+    definition[Definition store <br> libagent]--ApplicationReadyEvent-->boot;
+    etc[etc]--ApplicationReadyEvent-->boot;
+```
+
+##### lib/runtime
+
+A minimal spring boot application that provides the CftLibApi implementation and s2s simulator.
```

Comment:
Maybe could expand upon this to say what the CftLibApi is? Is it the entry point API that does the reverse proxying?